pkgs/http_profile/lib/http_profile.dart

```diff
@@ -14,20 +192,95 @@ final class HttpClientRequestProfile {
   static set profilingEnabled(bool enabled) =>
       HttpClient.enableTimelineLogging = enabled;
 
-  String? requestMethod;
-  String? requestUri;
+  final _data = Map<String, dynamic>();
+
+  /// The ID of the isolate the request was issued from.
+  String? get isolateId => _data['isolateId'] as String?;
+  set isolateId(String? value) {
+    _data['isolateId'] = value;
+    _updated();
+  }
+
+  /// The HTTP request method associated with the request.
+  String? get requestMethod => _data['requestMethod'] as String?;
+  set requestMethod(String? value) {
+    _data['requestMethod'] = value;
+    _updated();
+  }
+
+  /// The URI to which the request was sent.
+  String? get requestUri => _data['requestUri'] as String?;
+  set requestUri(String? value) {
+    _data['requestUri'] = value;
+    _updated();
+  }
+
+  /// The time at which the request was initiated, in microseconds.
+  int? get requestStartTimestamp => _data['requestStartTimestamp'] as int?;
+  set requestStartTimestamp(int? value) {
+    _data['requestStartTimestamp'] = value;
+    _updated();
+  }
+
+  /// The time at which the request was completed, in microseconds.
+  int? get requestEndTimestamp => _data['requestEndTimestamp'] as int?;
+  set requestEndTimestamp(int? value) {
+    _data['requestEndTimestamp'] = value;
+    _updated();
+  }
+
+  /// Details about the request.
+  set requestData(HttpProfileRequestData value) {
+    _data['requestData'] = value.toJson();
+    _updated();
+  }
+
+  StreamController<List<int>> _requestBody = StreamController<List<int>>();
```

Comment:
Ubernit: this can be `final _requestBody = ...`, so can `_responseBody` below.

Reply:
Done.